webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/campaigns/DataTypes.scala

```diff
@@ -84,14 +84,43 @@ object CampaignId                                                            {
   }
 }
 
+sealed trait CampaignStatusValue {
+  def value: String
+}
 @jsonDiscriminator("value")
-sealed trait CampaignStatus
+sealed trait CampaignStatus      {
+  def value: CampaignStatusValue
+}
+
+object CampaignStatus {
+  case object Enabled  extends CampaignStatusValue {
+    val value = "enabled"
+  }
+  case object Disabled extends CampaignStatusValue {
+    val value = "disabled"
+  }
+  case object Archived extends CampaignStatusValue {
+    val value = "archived"
+  }
+  val allValues = Enabled :: Disabled :: Archived :: Nil
```

Comment:
it can be replaced by a sealerate call to avoid unsync if value are added later on